R/growth.R

```diff
@@ -1902,24 +1910,62 @@ cleangrowth <- function(subjid,
     cat(sprintf("[%s] Re-centering data...\n", Sys.time()))
 
   # see function definition below for explanation of the re-centering process
-  # returns a data table indexed by param, sex, agedays
+  # returns a data table indexed by param, sex, agedays. can use NHANES reference
+  # data, derive from input, or use user-supplied data.
   if (!is.data.table(sd.recenter)) {
-    sd.recenter <- data.all[exclude < 'Exclude', sd_median(param, sex, agedays, sd.orig)]
-    if (sdmedian.filename != "") {
-      write.csv(sd.recenter, sdmedian.filename, row.names = F)
+    # Use NHANES medians if the string "nhanes" is specified instead of a data.table
+    # or if sd.recenter is not specified as "derive" and N < 5000.
+    if ((is.character(sd.recenter) & tolower(sd.recenter) == "nhanes") |
+      (!(is.character(sd.recenter) & tolower(sd.recenter) == "derive") & (data.all[, .N] < 5000))) {
+      nhanes_reference_medians_path <- ifelse(
+        ref.data.path == "",
+        system.file("extdata/nhanes-reference-medians.csv", package = "growthcleanr"),
```

Comment:
Should probably be changed to `file.path`, for more explicit applicability across different machines.

Reply:
Sure - I *think* we want to keep the rest of the `ifelse()` the same, e.g.:

```r
nhanes_reference_medians_path <- ifelse(
        ref.data.path == "",
        system.file(file.path("extdata", "nhanes-reference-medians.csv"), package = "growthcleanr"),
        paste(ref.data.path, "nhanes-reference-medians.csv", sep = "")
      )
```

Just want to be sure I'm not overthinking the `paste()` in the else slot. This already allows users to specify their own os-appropriate paths, and I don't think we want to change that.